Automattic/wp-calypso
# Add Second Show & Dismissal to Google My Business Nudge

This PR will change the behavior of the Google My Business Nudge to show itself two weeks after being dismissed.

## Specs

The Nudge should reappear for the last time 2 weeks after originally being dismissed.

## Testing Instructions

- Boot the branch locally or use calypso.live
- Go to the site stats for a business site which has a promote siteGoals
- In the Preference tool, make sure the `google-my-business-dismissible-nudge` is not there, delete it if it is.

![image](https://user-images.githubusercontent.com/230230/36608646-b5c4ac20-18ca-11e8-8a42-e82b293f8b7e.png)

- Notice the nudge appears when you load the Stats page
- Dismiss it with the cross in the top right corner
- The nudge should disappear
- Open redux dev tools to change the date of the preference by dispatching the following event:
```
{
  type: 'PREFERENCES_SET',
  key: 'google-my-business-dismissible-nudge',
  value: {
    'SITE_ID': [
      {
        dismissedAt: 1519407992903,
        type: 'dismiss'
      }
    ]
  }
}
```
(replace SITE_ID with your site id)
- It simulates the same action but make it so it happened 2 weeks ago.
- The nudge should appear again
- Dismiss it another time and reset the date again:
```
{
  type: 'PREFERENCES_SET',
  key: 'google-my-business-dismissible-nudge',
  value: {
    'SITE_ID': [
      {
        dismissedAt: 1519407992903,
        type: 'dismiss'
      },
      {
        dismissedAt: 1518524302835,
        type: 'dismiss'
      }
    ]
  }
}
```
(replace SITE_ID with your site id)
- The nudge should not appear (it has been dismissed 2 times)
- Reset the `google-my-business-dismissible-nudge` using the Preference menu
- Dismiss it by clicking on the "I already have a listing" button
- Dispatch the following action
```
{
  type: 'PREFERENCES_SET',
  key: 'google-my-business-dismissible-nudge',
  value: {
    'SITE_ID': [
      {
        dismissedAt: 1519407992903,
        type: 'already-listed'
      },
    ]
  }
}
```
(replace SITE_ID with your site id)
- The nudge should not appear
- Optional: go to another business site with siteGoals set to "promote". The nudge should appear

### Reviews
